openupgrade_scripts/scripts/account/15.0.1.2/pre-migration.py

```diff
@@ -1,16 +1,360 @@
+import logging
+
 from openupgradelib import openupgrade
 
+_logger = logging.getLogger(__name__)
+
+
+def convert_field_to_html(env):
+    openupgrade.convert_field_to_html(
+        env.cr, "res_company", "invoice_terms", "invoice_terms"
+    )
+    openupgrade.convert_field_to_html(env.cr, "account_fiscal_position", "note", "note")
+    openupgrade.convert_field_to_html(env.cr, "account_move", "narration", "narration")
+    openupgrade.convert_field_to_html(env.cr, "account_payment_term", "note", "note")
+
+
+def fast_fill_account_move_always_tax_exigible(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE account_move
+        ADD COLUMN IF NOT EXISTS always_tax_exigible boolean""",
+    )
+    openupgrade.logged_query(
```

Comment:
@tranngocson1996 code như bên dưới khá phức tạp
điều kiện nó đang là `not record.is_invoice(True) and not record._collect_tax_cash_basis_values()`
Vậy sẽ chia thành 3 trường hợp, 3 câu query cho đơn giản, không nên gộp thành 1 query
1. Nếu `record.is_invoice(True)` là `True` thì là `False`, tức là chỉ cần tìm với điều `move_type IN ['out_invoice', 'out_refund', 'in_refund', 'in_invoice', 'out_receipt', 'in_receipt']`
2. Nếu `record._collect_tax_cash_basis_values()` là `True` thì là `False`
3. Còn lại là `True`

Reply:
Ok em, cái nào hợp lý thì cứ chủ động làm nhé

Reply:
@royleviindoo dạ ok anh phần này em đã code và đẩy lên rồi anh ạ